### DISCUSSION TITLE
What is the recommended way to reuse SQL statements?

### DISCUSSION THREAD

**Discussion by oskarrough** [porsager/postgres#403] [Q&A] [answered]
Hello, was wondering what the recommended way to reuse SQL statements is here. I assumed `sql`${query}`` would work, but reading the docs I (maybe) I understand it's not meant to.

```js
const query = 'select * from posts'
await sql`${query}` // throws an error
await sql.unsafe(query) // works
```

To avoid using the `unsafe` method, would you wrap everything in functions?

```js
const queryById = (id) => sql`select * from posts where id = ${id}`
await queryById(42) // works
```

**Comment by porsager** [comment, marked as answer]
Oh yeah 😉 Functions Baby 🏋

**Reply by oskarrough** [reply]
tak!
